components/feature/downloads/src/main/java/mozilla/components/feature/downloads/AbstractFetchDownloadService.kt

```diff
@@ -86,8 +87,33 @@ abstract class AbstractFetchDownloadService : Service() {
         @GuardedBy("context") var status: DownloadJobStatus,
         var foregroundServiceId: Int = 0,
         var downloadDeleted: Boolean = false,
-        var notifiedStopped: Boolean = false
-    )
+        var notifiedStopped: Boolean = false,
+        @Volatile var notificationStatus: DownloadJobStatus? = null,
```

Comment:
Same as above, I don't think this needs to be volatile. `notificationStatus` is only ever read and written on the main thread. The `notificationUpdateScope` is the main scope. If not, we need to add a comment here explaining how there's concurrent access as it's not obvious :).